File: easy/149_juggling_zeros.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc, const char * argv[]) {
    FILE *file = fopen(argv[1], "r");
    char line[1024];
    char flag[3];
    char seq[50];

    while (fgets(line, 1024, file)) {
        char *p = line;
        unsigned int num = 0;
        while (sscanf(p, "%s %s", flag, seq) != EOF) {
            for (int i=0; i<strlen(seq); i++) {
                num <<= 1;
                if (strlen(flag) == 2)
                    num += 1;
            }
            p += strlen(flag) + strlen(seq) + 2;
        }
        printf("%ld\n", num);
    }

    return 0;
}

File: easy/149_juggling_zeros.py
import sys

#for line in open(sys.argv[1]):
#    it = iter(line.strip().split())
#    print(int(''.join(('1' if len(flag) == 2 else '0') * len(seq) for flag, seq in zip(it, it)), 2))

for line in open(sys.argv[1]):
    l = [len(s) for s in line.strip().split()]
    print(int(''.join(('0' if l[i] == 1 else '1') * l[i+1] for i in range(0, len(l), 2)), 2))
        
